### PR TITLE
Automates DB reset and migrations when unit and integration tests are…

### PR DESCRIPTION
**Pull request by yaoandrew**
This PR will automate database resets and migrations each time the integration and unit test suites are run. It uses a `testing.toml` and `vulcanize_testing` database for the tests similar to the way the mcd_transformers repo is setup.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,16 +41,34 @@ metalint: | $(METALINT)
...
+
 .PHONY: test
 test: | $(GINKGO) $(LINT)
 	go vet ./...
 	go fmt ./...
+	dropdb --if-exists $(TEST_DB)
+	createdb $(TEST_DB)
+	cd db/migrations;\
+    	$(GOOSE) postgres "$(TEST_CONNECT_STRING)" up
+	cd db/migrations/;\
```
Comment (rmulhol) [inline Makefile]: Any chance this works without these `cd` operations?

File: environments/testing.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [database]
-    name     = "vulcanize_private"
+    name     = "vulcanize_testing"
```
Comment (rmulhol) [inline environments/testing.toml]: Is this config used when we run the tests? If so, I'd be tempted to remove the `ipcPath` and verify that there's a warning to add one if you try to run the tests with it empty.

Also, I think such a warning does currently exist that points at `infura.toml` - probably worth making sure that's up to date if we're using this one now

File: go.mod [hunk 3]
```diff
@@ -44,6 +55,8 @@ require (
 	github.com/wsddn/go-ecdh v0.0.0-20161211032359-48726bab9208 // indirect
 	golang.org/x/net v0.0.0-20190603091049-60506f45cf65
 	golang.org/x/sync v0.0.0-20190423024810-112230192c58
+	google.golang.org/appengine v1.6.2 // indirect
+	gopkg.in/natefinch/npipe.v2 v2.0.0-20160621034901-c1b8fa8bdcce // indirect
```
Comment (rmulhol) [inline go.mod]: have you ran `go mod tidy`? Curious if we can omit some of these (especially since all of them need to be verified for compatibility with plugins)

File: environments/testing.toml (hunk 2 above)
Comment (m0ar) [reply environments/testing.toml]: Yees, warn for that! My tests fail regularly from me not setting that field. 

Would also be cool to prune some configs we're not using, since I'm always wondering which `toml` is going to be used for tests :woman_shrugging:

Comment (yaoandrew) [reply environments/testing.toml]: Added a warning when there is no `ipcPath` set in the file or via env variable. Using `vulcanize_testing` and removing `vulcanize_private` was an attempt to prune configs and make it consistent with mcdTransformers.

File: Makefile (hunk 1 above)
Comment (m0ar) [reply Makefile]: iirc the `cd` was a fix for the `-dir` flag not working as expected with `goose`, but I think in the end it was me being stupid: https://github.com/pressly/goose/issues/139

Comment (m0ar) [reply Makefile]: Short answer: yes, if you do it right :sweat_smile:

File: test_config/test_config.go [hunk 4]
```diff
@@ -44,13 +44,23 @@ func init() {
...
 	}
 	ipc := TestConfig.GetString("client.ipcPath")
+
+	// If we don't have an ipc path in the config file, check the env variable
+	if ipc == "" {
+		TestConfig.BindEnv("url", "INFURA_URL")
+		ipc = TestConfig.GetString("url")
+	}
+	if ipc == "" {
+		logrus.Fatal(errors.New("testing.toml IPC path or $INFURA_URL env variable need to be set"))
```
Comment (m0ar) [inline test_config/test_config.go]: :heart_eyes_cat:

Comment (m0ar) [inline test_config/test_config.go]: :heart_eyes_cat:
